code/modules/bitrunning/objects/gimmick_disks/sports_disk.dm

```diff
@@ -0,0 +1,142 @@
+
+/**
+ * Bitrunning gimmick loadouts themed around sports.
+ * Mostly for fun, have niche or little advantages.
+ */
+/obj/item/bitrunning_disk/gimmick/sports
+	name = "bitrunning gimmick: sports"
+	selectable_loadouts = list(
+		/datum/bitrunning_gimmick/boxer,
+		/datum/bitrunning_gimmick/skater,
+		/datum/bitrunning_gimmick/archer,
+		/datum/bitrunning_gimmick/fisher,
+		/datum/bitrunning_gimmick/gamer,
+	)
+
+
+/datum/bitrunning_gimmick/boxer
+	name = "Boxer"
+
+	granted_items = list(
+		/obj/item/clothing/gloves/boxing/evil,
+		/obj/item/clothing/under/shorts/red,
+		/obj/item/reagent_containers/condiment/protein,
+		/obj/item/reagent_containers/cup/glass/drinkingglass/filled/protein_blend,
+	)
+
+/obj/item/reagent_containers/cup/glass/drinkingglass/filled/protein_blend
+	name = "Protein Blend"
+	list_reagents = list(/datum/reagent/consumable/ethanol/protein_blend = 50)
+
+
+/datum/bitrunning_gimmick/skater
+	name = "Skater"
+
+	granted_items = list(
+		/obj/item/clothing/shoes/wheelys,
+		/obj/item/melee/skateboard,
+		/obj/item/clothing/suit/costume/wellworn_shirt/graphic,
+		/obj/item/clothing/head/soft/black,
+		/obj/item/clothing/shoes/sneakers/black,
+		/obj/item/storage/cans/sixenergydrink,
+	)
+
+/obj/item/storage/cans/sixenergydrink
+	name = "energy drink bottle ring"
+	desc = "Holds six energy drink cans. Remember to recycle when you're done!"
+
+	/// Pool of gamer drinks tm we may add from
+	var/list/gamer_drink_options = list(
+		/obj/item/reagent_containers/cup/soda_cans/space_mountain_wind = 50,
+		/obj/item/reagent_containers/cup/soda_cans/monkey_energy = 30,
+		/obj/item/reagent_containers/cup/soda_cans/volt_energy = 15,
+		/obj/item/reagent_containers/cup/soda_cans/thirteenloko = 5, 
+	)
+
+
+/datum/bitrunning_gimmick/archer
+	name = "Archer"
+
+	granted_items = list(
+		/obj/item/clothing/under/costume/kimono,
+		/obj/item/clothing/shoes/sandal/alt,
+		/obj/item/storage/bag/quiver/endless,
+		/obj/item/gun/ballistic/bow/longbow,
+		/obj/item/ammo_casing/arrow/holy/blazing,
+	)
+
+/obj/item/storage/bag/quiver/endless
+	name = "endless quiver"
+	desc = "Holds arrows for your bow. A deep digital void is contained within."
+
+/obj/item/storage/bag/quiver/endless/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_ATOM_EXITED, PROC_REF(handle_removed))
+
+/obj/item/storage/bag/quiver/endless/PopulateContents()
+	. = ..()
+	for(var/i in 1 to max_slots)
+		new arrow_path(src)
+
+/obj/item/storage/bag/quiver/endless/proc/handle_removed(datum/source, obj/item/gone)
```

Comment:
if you dump this does this cause an endless loop?

Reply:
so the answer is _no_ but it does allow for dumping a fucktonne of arrows at once so like. yeah I gotta change that up- probably just like. lower it to holding only one arrow. not like you actually _need_ 40 at once